go/vt/sqlparser/comments.go

```diff
@@ -349,6 +569,21 @@ func AllowScatterDirective(stmt Statement) bool {
 	return checkDirective(stmt, DirectiveAllowScatter)
 }
 
+// ForeignKeyChecksState returns the state of foreign_key_checks variable if it is part of a SET_VAR optimizer hint in the comments.
+func ForeignKeyChecksState(stmt Statement) FkChecksState {
+	cmt, ok := stmt.(Commented)
+	if ok {
+		fkChecksVal := cmt.GetParsedComments().GetMySQLSetVarValue(sysvars.ForeignKeyChecks.Name)
+		switch strings.ToLower(fkChecksVal) {
+		case "on", "1":
+			return FkChecksOn
+		case "off", "0":
+			return FkChecksOff
```

Comment:
we don't want to fail for weird values here? WWMD? (What Would MySQL Do?)

Reply:
That is exactly what I did 😄 

```sql
mysql [localhost:8032] {msandbox} (foreign_key_rfc) > insert into fk_t16 values (5, 34);
ERROR 1452 (23000): Cannot add or update a child row: a foreign key constraint fails (`foreign_key_rfc`.`fk_t16`, CONSTRAINT `fk_t16_ibfk_1` FOREIGN KEY (`col`) REFERENCES `fk_t15` (`col`) ON DELETE CASCADE ON UPDATE CASCADE)
mysql [localhost:8032] {msandbox} (foreign_key_rfc) > insert /*+ SET_VAR(foreign_key_checks=O) */ into fk_t16 values (5, 34);
ERROR 1452 (23000): Cannot add or update a child row: a foreign key constraint fails (`foreign_key_rfc`.`fk_t16`, CONSTRAINT `fk_t16_ibfk_1` FOREIGN KEY (`col`) REFERENCES `fk_t15` (`col`) ON DELETE CASCADE ON UPDATE CASCADE)
mysql [localhost:8032] {msandbox} (foreign_key_rfc) > insert /*+ SET_VAR(foreign_key_checks=On) */ into fk_t16 values (5, 34);
ERROR 1452 (23000): Cannot add or update a child row: a foreign key constraint fails (`foreign_key_rfc`.`fk_t16`, CONSTRAINT `fk_t16_ibfk_1` FOREIGN KEY (`col`) REFERENCES `fk_t15` (`col`) ON DELETE CASCADE ON UPDATE CASCADE)
mysql [localhost:8032] {msandbox} (foreign_key_rfc) > insert /*+ SET_VAR(foreign_key_checks=Off) */ into fk_t16 values (5, 34);
Query OK, 1 row affected (0.00 sec)

mysql [localhost:8032] {msandbox} (foreign_key_rfc) > select @@foreign_key_checks;
+----------------------+
| @@foreign_key_checks |
+----------------------+
|                    1 |
+----------------------+
1 row in set (0.00 sec)

mysql [localhost:8032] {msandbox} (foreign_key_rfc) > set @@foreign_key_checks=off;
Query OK, 0 rows affected (0.00 sec)

mysql [localhost:8032] {msandbox} (foreign_key_rfc) > select @@foreign_key_checks;
+----------------------+
| @@foreign_key_checks |
+----------------------+
|                    0 |
+----------------------+
1 row in set (0.00 sec)

mysql [localhost:8032] {msandbox} (foreign_key_rfc) > insert /*+ SET_VAR(foreign_key_checks=On) */ into fk_t16 values (6, 34);
ERROR 1452 (23000): Cannot add or update a child row: a foreign key constraint fails (`foreign_key_rfc`.`fk_t16`, CONSTRAINT `fk_t16_ibfk_1` FOREIGN KEY (`col`) REFERENCES `fk_t15` (`col`) ON DELETE CASCADE ON UPDATE CASCADE)
mysql [localhost:8032] {msandbox} (foreign_key_rfc) > insert /*+ SET_VAR(foreign_key_checks=O) */ into fk_t16 values (6, 34);
Query OK, 1 row affected, 1 warning (0.01 sec)
```

For a value that MySQL doesn't understand, it just treats as though it was unspecified. We do the same thing. I considered throwing an error, but chose to emulate MySQL.

Reply:
I'll add a comment in the code to call out this behaviour explicitly.

Reply:
Done!